### PR TITLE
 jsonnet/telemeter/server: move whitelist out of secret

### PR DESCRIPTION
**Pull request by squat**
This commit removes the whitelist from the Telemeter server secret and
places it instead in a series of command line `--whitelist` flags. This
is helpful for several reasons:
1. as the whitelist is no longer in a secret it no longer has to be
handled manually, preventing potential human error when the whitelist
changes; and
2. when the whitelist changes, the DaemonSet spec will also change so
the new configuration will automatically be rolled out without needing
to manually kick pods.

cc @s-urbaniak @jfchevrette

### CODE REVIEW

File: jsonnet/telemeter/server/kubernetes.libsonnet [hunk 1]
```diff
@@ -50,9 +48,13 @@ local whitelistFileName = 'whitelist';
       local rhdUsername = containerEnv.fromSecretRef('RHD_USERNAME', secretName, 'rhd.username');
       local rhdPassword = containerEnv.fromSecretRef('RHD_PASSWORD', secretName, 'rhd.password');
       local rhdClientID = containerEnv.fromSecretRef('RHD_CLIENT_ID', secretName, 'rhd.client_id');
-      local secretMount = containerVolumeMount.new(secretVolumeName, secretMountPath);
       local secretVolume = volume.fromSecret(secretVolumeName, secretName);
 
+      local whitelist = std.map(
+        function(rule) "--whitelist='%s'" % std.strReplace(rule, 'ALERTS', 'alerts'),
```
Comment (s-urbaniak) [inline jsonnet/telemeter/server/kubernetes.libsonnet]: does it make sense to do a std.asciiLower(str) here instead?

Comment (squat) [reply jsonnet/telemeter/server/kubernetes.libsonnet]: no it does not make sense here; there may very well be metric names or labels in the whitelist that we do _not_ want completely lowercased. The only one we care about today is the `ALERTS` metric name.

Comment (s-urbaniak) [reply jsonnet/telemeter/server/kubernetes.libsonnet]: agreed

**Comment by squat** [comment]
fixes #121

**Pull request merged**